cmd/thermal-recorder/main.go

```diff
@@ -322,3 +344,47 @@ func writeInitialFramesToFile(writer *cptv.FileWriter, frames []*lepton3.Frame,
 
 	return nil
 }
+
+func newSnapshot() error {
+	if snapshotPending.Load() == true {
+		//TODO return the pending snapshot error also.
+		return errors.New("snapshot already pending")
```

Comment:
It's probably ok to just return `nil` here. If a snapshot is already being produced then that's what the caller wanted right?